### ISSUE TITLE
Refactor Verifier to builder approach

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#5186]
There are too many overloads in `Verifier` with too many arguments and duplications. We should refactor it to a builder approach instead. 

The exact approach and API shape will be discussed.

### Original proposal by @Corniel in #4971

The SonarAnalyzer TestFramework is awesome. It contains a lot of options to pick when verifying an analyzer.

However, given the enormous number of overloads makes it a bit clumsy. By providing a builder usage could improve:

``` C#
[TestMethod]
public void BreakOutsideSwitch() =>
    new BreakOutsideSwitch().AddSource(@"TestCases\BreakOutsideSwitch.cs").Verify();
```

And a more complex example:

``` C#
[TestMethod]
public void DeclareTypesInNamespaces_CS_Before8() =>
    new CS.DeclareTypesInNamespaces()
    .AddSource(@"TestCases\DeclareTypesInNamespaces.BeforeCSharp8.cs")
    .WithOptions(ParseOptionsHelper.BeforeCSharp8)
    .Verify();
```

**Comment by Corniel** [comment]
I made a version that I myself for testing my own rules. That has no Sonar dependency, but therefor lacks some of the features you (probably still) need. Also, I handle secondary and primary issues a bit different I think.

What I also did, I always log all issues to the console, both the expected, missing, as extra. Made my live easier.

Anyway, feel inspired:
https://github.com/Corniel/roslyn-test-tools
